### PR TITLE
Introduce material enumeration

### PR DESCRIPTION
**Pull request by fbtom**
Introduce Material.
Create tests 

Issue: #22

### CODE REVIEW

File: src/app/warehouse/material.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+enum class Material : std::uint8_t
+{
+    DentalCement = 0,
+    Amalgam = 1,
+    DentalComposite = 2,
+    Size = 255
+};
+
+inline std::string convertMaterial(const Material &material)
```
Comment (BaqWin) [inline src/app/warehouse/material.hpp:13]: What if we pass null as an argument?

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

File: src/app/warehouse/material.hpp (hunk 1 above)
Comment (fbtom) [reply src/app/warehouse/material.hpp:13]: C++ strong typing ensures that we cannot pass null as an argument.

**Comment by fbtom** [comment]
deprecated 
due to wrong names on chart material was added as issue

**Review by mBialczak — COMMENTED** [review]
General CmakeLists files analyze suggest validness of making two main trees of CMakeLists.txt files: one for main app, one for tests.
In my opinion, they should not be leaves of main CmakeLists as it is now.
An Issue will be created.

**Comment by ziobron** [comment]
Thanks @mBialczak for Code Review.
🏅 1 XP granted

**Pull request closed without merging**